MomentumDiscordBot/Commands/General/GeneralModule.cs

```diff
@@ -54,8 +54,36 @@ public class GeneralModule : MomentumModuleBase
                 if (Uri.IsWellFormedUriString(command.ButtonUrl, UriKind.Absolute))
                     message.AddComponents(new DiscordLinkButtonComponent(command.ButtonUrl,
                         command.ButtonLabel ?? "Link"));
-                await context.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource,
-                    new DiscordInteractionResponseBuilder(message));
+                if (ulong.TryParse(replyMessageId, out ulong id))
+                {
+                    DiscordMessage replyMessage;
+                    try
+                    {
+                        // check if the selected message is from this channel
+                        replyMessage = await context.Channel.GetMessageAsync(id);
+                    }
+                    catch (DSharpPlus.Exceptions.NotFoundException)
+                    {
+                        await context.CreateResponseAsync(new DiscordEmbedBuilder()
+                        {
+                            Title = $"Can't find message {id} in this channel.",
+                            Color = MomentumColor.Red
+                        }, true);
+                        return;
+                    }
+                    await context.Channel.SendMessageAsync(message.WithReply(id, true));
+                    await context.CreateResponseAsync(new DiscordEmbedBuilder()
```

Comment:
I don't think this message is really necessary. If the command is successful you'll see the message that's printed, so this seems redundant.